src/features/events/components/EventContactCard.tsx

```diff
@@ -0,0 +1,129 @@
+import CloseIcon from '@mui/icons-material/Close';
+import FaceOutlinedIcon from '@mui/icons-material/FaceOutlined';
+import FaceRetouchingOffOutlinedIcon from '@mui/icons-material/FaceRetouchingOffOutlined';
+import { Box, Button, Typography } from '@mui/material';
+import { FC, useContext } from 'react';
+
+import EventDataModel from 'features/events/models/EventDataModel';
+import messageIds from 'features/events/l10n/messageIds';
+import { useMessages } from 'core/i18n';
+import { ZetkinEvent } from 'utils/types/zetkin';
+import ZUIAvatar from 'zui/ZUIAvatar';
+import ZUICard from 'zui/ZUICard';
+import { ZUIConfirmDialogContext } from 'zui/ZUIConfirmDialogProvider';
+import { MUIOnlyPersonSelect as ZUIPersonSelect } from 'zui/ZUIPersonSelect';
+
+interface EventContactCardProps {
+  data: ZetkinEvent;
+  model: EventDataModel;
+  orgId: number;
+}
+
+const EventContactCard: FC<EventContactCardProps> = ({
+  data,
+  model,
+  orgId,
+}) => {
+  const messages = useMessages(messageIds);
+  const { showConfirmDialog } = useContext(ZUIConfirmDialogContext);
+
+  const handleSelectedPerson = (personId: number) => {
+    model.setContact(personId);
+  };
+
+  return (
+    <Box
+      mb={2}
+      sx={{
+        '& div': {
+          flexDirection: 'column',
+          flexWrap: 'wrap',
+        },
+        display: 'block',
+      }}
+    >
+      <ZUICard
+        header={
+          <Box>
+            {model.getData().data?.contact?.id ? (
+              <Box>
+                <FaceOutlinedIcon sx={{ margin: 1, verticalAlign: 'middle' }} />
+                {messages.eventContactCard.header()}
+              </Box>
+            ) : (
+              <Box>
+                <FaceRetouchingOffOutlinedIcon
+                  sx={{ margin: 1, verticalAlign: 'middle' }}
+                />
+                {messages.eventContactCard.noContact()}
+              </Box>
+            )}
+          </Box>
+        }
+        status={
+          <Box>
+            {data.contact?.id ? (
+              <>
+                <Box
+                  m={1}
+                  sx={{ display: 'inline-block', verticalAlign: 'middle' }}
+                >
+                  <ZUIAvatar orgId={orgId} personId={data.contact.id} />
+                </Box>
+                <Typography
+                  sx={{ display: 'inline-block', verticalAlign: 'middle' }}
+                >
+                  {data.contact.name}
+                </Typography>
+              </>
+            ) : (
+              <Box sx={{ display: 'grid' }}>
+                <Box
+                  m={1}
+                  mb={3}
+                  mt={3}
+                  sx={{
+                    '& input': {
+                      alignSelf: 'self-start',
+                      minWidth: '100px !important',
+                      width: '200px !important',
+                    },
```

Comment:
Why is this necessary? I'm guessing that this is what's causing the person select widget to be too wide on narrow screens (see screenshot). This seems like a hack to me, but I understand that there was probably some issue without it that you wanted to solve. I'm curious what that was, maybe we can find a simpler solution?

![image](https://user-images.githubusercontent.com/550212/235184311-1a0865c3-3b9e-4afc-9ee6-05e6df15142a.png)

Reply:
Yes! The input from the `ZUIPersonSelect `has by default a _min-width: 30px_ so this was the only way I found to make the placeholder display the content and override the default min-width, otherwise:

![image](https://user-images.githubusercontent.com/36491300/235691321-1f26c237-7ca7-453c-9004-0e5864542287.png)

But to avoid this behavior I can have a min-width of 100px for example to don't let the input be cutted?

Reply:
Lets find some time tomorrow to look at this together. It should not be possible to employ hacks like this. Instead, lets find a nice way to control the min width from outside the component (if at all needed).